cmake/cuda.cmake

```diff
@@ -56,6 +56,8 @@ else()
         HINT ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
 endif()
 
+find_package(NVTX REQUIRED)
```

Comment:
Can we guarantee all systems will have NVTX ? If not, should we maybe allow disabling this ?

Reply:
CUDA is usually installed monolithically, always including NVTX. On Windows, the location may differ for the legacy NVTX, but for NVTX3, it is part of the include folder, so it should work everywhere

Reply:
Are you sure ? AFAIK, they also separated into the core (compilers) and libraries (cublas and cusparse). In particular, I have had issues with the nvhpc compiler suite, where they usually do this separation and you need to go looking for where cublas and cusparse are installed (if they are)

Reply:
Good point! In the tarball I downloaded, nvtx is grouped with `cuda`, not `math_libs`, so it seems to be in the right place?

Reply:
If it is indeed always grouped with CUDA, then it should be fine.